geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/Strategy.scala

```diff
@@ -163,6 +160,22 @@ trait Strategy {
       configureFeatureType(cfg, featureType)
 
       Some(cfg)
-    } else None
+    case mapAggregationQuery if mapAggregationQuery.getHints.containsKey(MAP_AGGREGATION_KEY) =>
```

Comment:
I'm thinking that we may be served by a general getQueryHints which returns an object to case match on.  I suppose I'm trying to hide the 'if's a little bit.